src/frontends/tensorflow_common/include/helper_ops/switch.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "internal_operation.hpp"
+#include "openvino/frontend/exception.hpp"
+#include "tf_utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+
+// Internal operation for Switch
+// Forwards data to the output port determined by predicate
+// It has two outputs:
+// 1. output_false - forward data to this output in case predicate to be false
+// 2. output_true - forward data to this output in case predicate to be true
+class Switch : public InternalOperation {
+public:
+    OPENVINO_OP("Switch", "ov::frontend::tensorflow", InternalOperation);
+
+    Switch(const Output<Node>& data,
+           const Output<Node>& pred,
+           uint32_t switch_marker,
+           const std::shared_ptr<DecoderBase>& decoder = std::make_shared<DecoderFake>())
+        : InternalOperation(decoder, OutputVector{data, pred}, 2, "Switch"),
+          m_switch_marker(switch_marker) {
+        validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        auto data_type = get_input_element_type(0);
+        auto data_shape = get_input_partial_shape(0);
+
+        set_output_type(0, data_type, data_shape);
+        set_output_type(1, data_type, data_shape);
+    }
+
+    bool is_cond_flow_marker_set() const {
+        const auto this_node = this->shared_from_this();
+        if (!cf_marker_exists(this_node)) {
+            return false;
+        }
+
+        auto cf_marker = get_cf_marker(this_node);
+        return cf_marker.new_markers.size() == 1;
```

Comment:
Could you explain how it works, please? As I can see in code when we are filling new_markers - we expect it could be > 1, when we use it in Switch - we expect exactly one. Is it some helpers for further features?

Reply:
Yes, it is reserved for future support of `CaseSwitch` feature. And moreover, it is more convenient to use vector because it is not always that some Node has new_marker.